clang/include/clang/AST/Type.h

```diff
@@ -2696,6 +2699,8 @@ class alignas(TypeAlignment) Type : public ExtQualsTypeCommonBase {
     return static_cast<TypeDependence>(TypeBits.Dependence);
   }
 
+  const TemplateDecl *getSpecializedTemplateDecl() const;
```

Comment:
Does that need to be in the header? (the implementation only work for type, it's not exactly generic)